substrate/frame/broker/src/utility_impls.rs

```diff
@@ -118,4 +118,33 @@ impl<T: Config> Pallet<T> {
 
 		Ok(Some((region_id, region)))
 	}
+
+	// Remove a region from on-demand pool contributions. Useful in cases where it was pooled
+	// provisionally and it is being redispatched (partition/interlace/assign).
+	//
+	// Takes both the region_id and (a reference to) the region as arguments to avoid another DB
+	// read. No-op for regions which have not been pooled.
+	pub(crate) fn force_unpool_region(region_id: RegionId, region: &RegionRecordOf<T>) {
+		// We don't care if this fails or not, just that it is removed if present. This is to
+		// account for the case where a region is pooled provisionally and redispatched.
+		if let Some(_) = InstaPoolContribution::<T>::take(region_id) {
+			let current_timeslice = Self::current_timeslice();
```

Comment:
Using here the current timeslice is incorrect? This timeslice was already scheduled and trying to change this will not change anything? We should use `last_committed_timeslice + 1` as the timeslice to process.

Reply:
It took me a while to decide which timeslice to use, I thought this would be the best one as it's the next one to be committed, but actually it might already have been committed. But what you're saying makes sense as the tick might have already run for this timeslice and processed the contributions.

Reply:
We are also committing `advance_notice` blocks before, thus, `current_slice()` will still return the old timeslice for `advance_notice` blocks.

Reply:
OK it actually needs to be `current_timeslice() + 2` 😳
It makes sense when you think about it though:
The tick that happens in timeslice N processes the history for [timeslice N+1](https://github.com/paritytech/polkadot-sdk/blob/donal-broker-force-unpool/substrate/frame/broker/src/tick_impls.rs#L55) and so when we want to force unpool in timeslice N, we need to register the change in `InstaPoolIo` in timeslice N+2 or the tick never sees the change and the `InstaPoolHistory` keeps them forever.

We could also use `last_committed_timeslice + 1` but is it worth the two db reads (config and status) for the sake of sometimes working 1 timeslice sooner when they beat the tick?